plugins/inputs/snmp_trap/snmp_trap.go

```diff
@@ -155,6 +162,30 @@ func makeTrapHandler(s *SnmpTrap) handler {
 		tags["version"] = packet.Version.String()
 		tags["source"] = addr.IP.String()
 
+		if packet.Version == gosnmp.Version1 {
+			// Follow the procedure described in RFC 2576 3.1 to
+			// translate a v1 trap to v2.
+			var trapOid string
+
+			if packet.GenericTrap > 0 && packet.GenericTrap < 6 {
+				trapOid = "1.3.6.1.6.3.1.1.5." + strconv.Itoa(packet.GenericTrap+1)
+			} else if packet.GenericTrap == 6 {
+				trapOid = packet.Enterprise + ".0." + strconv.Itoa(packet.SpecificTrap)
+			}
+
+			if trapOid != "" {
+				e, err := s.lookup(trapOid)
+				if err != nil {
+					s.Log.Errorf("Error resolving V1 OID: %v", err)
+					return
+				}
+				setTrapOid(tags, trapOid, e)
+			}
+			tags["agent_address"] = packet.AgentAddress
```

Comment:
Let's set this as the "source" tag.  Possibly in the future this could be the `snmpTrapAddress.0` if we decide we should act as a proxy but I don't feel confident enough now that we will want to do this.

Reply:
I'm not sure it's always valid.  After chatting with you we decided to continue using addr.IP as the "source" tag and validate agent address before adding it as a tag.